### PR TITLE
Migrate DirectedGraphSpec from spec to ScalaTest

### PR DESCRIPTION
**Pull request by adamkozuch**
My first migrated test

### CODE REVIEW

File: cassovary-core/src/test/scala/com/twitter/cassovary/graph/DirectedGraphSpec.scala [hunk 1]
```diff
@@ -13,58 +13,60 @@
...
     "maxNodeId should be correct" in {
-      graph.maxNodeId mustEqual 2
+      twoNodeGraph.maxNodeId shouldEqual 2
     }
 
     "elements should return all the nodes" in {
-      graph.iterator.toList.map { _.id }.sortBy (+_) mustEqual List(1, 2)
+      twoNodeGraph.iterator.toList.map {
+        _.id
+      }.sortBy(+_) shouldEqual List(1, 2)
```
Comment (szymonm) [inline cassovary-core/src/test/scala/com/twitter/cassovary/graph/DirectedGraphSpec.scala]: What's the point in sortBy(+_). Isn't sorted better? Or, even better, toSet shouldEqual Set(1,2)

File: cassovary-core/src/test/scala/com/twitter/cassovary/graph/DirectedGraphSpec.scala [hunk 2]
```diff
@@ -13,58 +13,60 @@
...
 
     "maxNodeId should be correct" in {
-      graph.maxNodeId mustEqual 2
+      twoNodeGraph.maxNodeId shouldEqual 2
     }
 
     "elements should return all the nodes" in {
-      graph.iterator.toList.map { _.id }.sortBy (+_) mustEqual List(1, 2)
+      twoNodeGraph.iterator.toList.map {
+        _.id
```
Comment (szymonm) [inline cassovary-core/src/test/scala/com/twitter/cassovary/graph/DirectedGraphSpec.scala]: I don't like new lines here. I would write it in one line as there is no serious logic in the _.id. And probably only in ( ).

File: cassovary-core/src/test/scala/com/twitter/cassovary/graph/DirectedGraphSpec.scala [hunk 3]
```diff
@@ -13,58 +13,60 @@
...
 
     "maxNodeId should be correct" in {
-      graph.maxNodeId mustEqual 15
+      sixNodeGraph.maxNodeId shouldEqual 15
     }
 
     "elements should return all the nodes" in {
-      graph.iterator.toList.map { _.id }.sortBy(+_) mustEqual List(10, 11, 12, 13, 14, 15)
+      sixNodeGraph.iterator.toList.map {
+        _.id
```
Comment (szymonm) [inline cassovary-core/src/test/scala/com/twitter/cassovary/graph/DirectedGraphSpec.scala]: as before

**Comment by szymonm** [comment]
Hi @adamkozuch, I'm glad to see your change.

This test, however, was a bit broken from the beginning. The specification written as strings does not fit well to the WordSpec schema. This results in output like:
```
[info] + two node graph with each following the other should
[info]   + nodeCount and edgeCount should be correct
[info]   + maxNodeId should be correct
[info]   + elements should return all the nodes
[info]   + getNodeById should work in both positive and negative cases
```
that is not fully language correct. I think it's possible to fix it using WordSpec 'when' special word. You can check the documentation here: http://doc.scalatest.org/1.9.1/#org.scalatest.WordSpec

Could you try to fix this issues?

**Comment by adamkozuch** [comment]
I made changes. Is it correct now?

**Comment by szymonm** [comment]
Thanks, LGTM @pankajgupta

**Pull request merged**